core/src/crds_gossip_push.rs

```diff
@@ -135,16 +135,19 @@ impl CrdsGossipPush {
         );
 
         let mut keep = HashSet::new();
-        let mut peer_stake_sum = 0;
+        let mut peer_stake_sum = *stakes.get(origin).unwrap_or(&0);
+        keep.insert(*origin);
         for next in shuffle {
-            let (next_peer, next_stake) = staked_peers[next];
-            keep.insert(next_peer);
-            peer_stake_sum += next_stake;
             if peer_stake_sum >= prune_stake_threshold
                 && keep.len() >= CRDS_GOSSIP_PRUNE_MIN_INGRESS_NODES
```

Comment:
will always inserting `origin` into `keep` affect the diversity of the prunes here? Since we always keep at most `CRDS_GOSSIP_PRUNE_MIN_INGRESS_NODES` unpruned, should it be `CRDS_GOSSIP_PRUNE_MIN_INGRESS_NODES + 1` now?

Reply:
@carllin that is a good point. I bumped up `CRDS_GOSSIP_PRUNE_MIN_INGRESS_NODES` by one to account for the `origin`. I did some testings on gce, and this is still better than the current code.